### PR TITLE
Separate VisualStudio projects from main solution

### PR DESCRIPTION
**Pull request by Jklawreszuk**
# PR Details

## Related Issue

Solves issue #1671 

## Types of changes

- [x] Docs change / refactoring / dependency upgrade

### CODE REVIEW

**Comment by manio143** [comment]
Would be good to have a look at how the TeamCity builds the VS Plugin for publishing.
Another idea is to create a solution filter for the editor which would exclude the plugin for those interested in the editor only (with appropriate docs change).

**Comment by Jklawreszuk** [comment]
@manio143 To be honest, I do not have any experience using TeamCity, but here is (propably) Stride's TeamCity pipeline diagram:

image

First, It runs "Package" target
```bash
MSBuild.exe Stride.build -t:Package @C:\BuildAgent\temp\agentTmp\1.rsp /m /nr:false /p:StridePlatforms="Windows;Android;iOS;Linux;macOS" /p:StrideGraphicsApiDependentBuildAll=true /p:StrideEditorTargetFrameworks="net472;net5.0-windows" /p:StrideSign="true" /p:StrideSignUser="sc-stride@dotnetfoundation.org" /p:StrideSignSecret="*******"
```
And If we take a look into stride.buid file, "Package" target executes other targets i.e. changed VSIXPlugin target:
```xml
<Target Name="Package" DependsOnTargets="PreparePackageBuild;$(_StridePackageTargetPrerequisitesInstaller);Build;SignPackage;VSIXPlugin"/>
```

Finally, TC is deploying vsix packages 
```bash
MSBuild.exe Stride.build /t:PublishVSIX @C:\BuildAgent\temp\agentTmp\1.rsp /p:StrideStoreUrl=https://api.nuget.org/v3/index.json /p:StrideStoreApiKey=*******
```
and definition:
```xml
  <Target Name="PublishVSIX">
    <Error Condition="'$(StrideStoreApiKey)' == ''" Text="Missing env variable StrideStoreApiKey"/>
    <Exec Command='"$(NuGetPath)" push ..\bin\vsix\*.nupkg -ApiKey $(StrideStoreApiKey) -Source $(StrideStoreUrl) -NoSymbols'/>
  </Target>
  ```
  
**TLDR:** Imho, It should work 😅. Question I have is using solution filter more beneficial than just move projects to solution file?

**Comment by manio143** [comment]
@Jklawreszuk Your explanation is great and I think it's fine to do it like this 😊
Since very few devs would compile the VS Plugin locally I think it makes sense to have its own solution.

**Comment by Eideren** [comment]
Alright, let's merge this in then :)
No objections @manio143 / @Jklawreszuk ?

**Review by manio143 — APPROVED** [review]

**Comment by Jklawreszuk** [comment]
@Eideren No, I think it should be merged then :)

**Comment by xen2** [comment]
We can do that as a later step, but I remember checking about .slnf (Solution Filter) and thinking they would be good for such use case: no need to maintain several .sln, and it's easy to load everything when doing refactors.

I remember postponing it because MSBuild/Teamcity didn't support it well at the time, but it might be better now.

**Comment by Jklawreszuk** [comment]
@xen2 But my intention was to logically separate the actual engine compilation process from the plugin compilation for Visual Studio, I'm not sure if the solution filter solves that. So I decided to create another sln project - yes, it makes another solution to maintain, but this approach is clear, and current contributors rather rarely look at this code anyway. I'm not saying the solution filter is a bad idea, but I just don't see it

**Comment by tebjan** [comment]
@Jklawreszuk this PR appears to have led to the Stride.VisuallStudio nugets are not being released anymore. but @xen2 just pushed a quick fix...
